File: 4/1.py
import re

with open("input", "r") as file:
    text = file.read().split('\n\n')
    text = [i.replace('\n', ' ') for i in text]
    text = [i.replace(' ', ':') for i in text]
    text = [i.split(':')[::2] for i in text]

ans = 0
for passport in text:
    if len(passport) == 8 or (len(passport) == 7 and 'cid' not in passport):
        ans+=1

print(ans)

File: 4/1.rs
use std::fs::read_to_string;

fn main() {
    let input = read_to_string("./input").unwrap();

    let vec_input: Vec<_> = input
        .split("\n\n")
        .map(|x| x
            .replace('\n', " ")
            .replace(' ', ":")
        ).collect();

    let mut ans = 0;
    for i in vec_input.iter() {
        let fields: Vec<_> = i.split(":").step_by(2).collect();
        
        let cid_exists = fields.contains(&"cid");
        let len = fields.len();
        
        if len == 8 || (len == 7 && !cid_exists) {
            ans += 1;
        }
    }

    println!("{}", ans);
}